### DISCUSSION TITLE
order status

### DISCUSSION THREAD

**Discussion by huwgit** [vendurehq/vendure#1933] [Q&A]
I need to add ORDER PICKED as a status - i.e all items have been collected and are ready for shipping (which could be in 2 weeks). What is the best way to do this. Also at this point is part payment of an order possible before shipping.

**Comment by michaelbromley** [comment]
Hi!
yes, this is covered in the docs here: https://www.vendure.io/docs/developer-guide/customizing-the-order-process/

As for part payment - you'll need to create a custom PaymentMethodHandler I think, which supports the exact payment workflow you require. Docs here: https://www.vendure.io/docs/developer-guide/payment-integrations/

**Reply by huwgit** [reply]
I copied the example to test it but the Admin UI will not allow manually transition...

// customer-validation-process.ts
import { CustomOrderProcess } from '@vendure/core';


declare module '@vendure/core' {
    interface CustomOrderStates {
      ValidatingCustomer: never;
    }
  }
  
export const customerValidationProcess: CustomOrderProcess<'ValidatingCustomer'> = {
  transitions: {
    AddingItems: {
      to: ['ValidatingCustomer'],
      mergeStrategy: 'replace',
    },
    ValidatingCustomer: {
      to: ['ArrangingPayment', 'AddingItems'],
    },
  },
};

and 

// vendure-config.ts
import { VendureConfig } from '@vendure/core';
import { customerValidationProcess } from './customer-validation-process';

export const config: VendureConfig = {
  // ...
  orderOptions: {
    process: [customerValidationProcess],
  },
};

**Reply by michaelbromley** [reply]
What happens?

**Reply by huwgit** [reply]
it runs as normal on npm run dev. In Admin UI / orders it does not give an option to transition to it from Adding Items state.
